src/components/expanded-state/custom-gas/FeesPanel.js

```diff
@@ -0,0 +1,509 @@
+        return;
+
+      setCustomMaxPriorityFee(get(newMaxPriorityFeePerGas, 'gwei', 0));
+      setCustomMaxBaseFee(parseInt(get(newMaxFeePerGas, 'gwei', 10)));
+
+      const newGasParams = {
+        ...selectedGasFee.gasFeeParams,
+        maxFeePerGas: newMaxFeePerGas,
+        maxPriorityFeePerGas: newMaxPriorityFeePerGas,
+      };
+      updateToCustomGasFee(newGasParams);
+    },
+    [selectedGasFee.gasFeeParams, updateToCustomGasFee]
+  );
+
+  const addMinerTip = useCallback(() => {
+    updateGasFee({
+      priorityFeePerGas: calculateMinerTipAddDifference(maxPriorityFee),
+    });
+  }, [updateGasFee, maxPriorityFee]);
+
+  const substMinerTip = useCallback(() => {
+    updateGasFee({
+      priorityFeePerGas: -calculateMinerTipSubstDifference(maxPriorityFee),
+    });
+  }, [updateGasFee, maxPriorityFee]);
+
+  const addMaxFee = useCallback(() => {
+    updateGasFee({ feePerGas: GAS_FEE_INCREMENT });
+  }, [updateGasFee]);
+
+  const substMaxFee = useCallback(
+    () => updateGasFee({ feePerGas: -GAS_FEE_INCREMENT }),
+    [updateGasFee]
+  );
+
+  const onMaxBaseFeeChange = useCallback(
+    ({ nativeEvent: { text } }) => {
+      text = text || 0;
+      const maxFeePerGas = parseGasFeeParam(gweiToWei(text));
+      const gweiMaxFeePerGas = maxFeePerGas?.gwei || 0;
+
+      const newGweiMaxFeePerGas = Math.round(gweiMaxFeePerGas * 100) / 100;
+
+      const newMaxFeePerGas = parseGasFeeParam(gweiToWei(newGweiMaxFeePerGas));
+
+      if (newMaxFeePerGas.amount < 0) return;
+
+      setCustomMaxBaseFee(text);
+
+      const newGasParams = {
+        ...selectedGasFee.gasFeeParams,
+        maxFeePerGas: newMaxFeePerGas,
+      };
+      updateToCustomGasFee(newGasParams);
+    },
+    [selectedGasFee.gasFeeParams, updateToCustomGasFee]
+  );
+
+  const onMinerTipChange = useCallback(
+    ({ nativeEvent: { text } }) => {
+      text = text || 0;
+      const maxPriorityFeePerGas = parseGasFeeParam(gweiToWei(text));
+      const gweiMaxPriorityFeePerGas = maxPriorityFeePerGas?.gwei || 0;
+
+      const newGweiMaxPriorityFeePerGas =
+        Math.round(gweiMaxPriorityFeePerGas * 100) / 100;
+
+      const newMaxPriorityFeePerGas = parseGasFeeParam(
+        gweiToWei(newGweiMaxPriorityFeePerGas)
+      );
+      if (newMaxPriorityFeePerGas.amount < 0) return;
+
+      setCustomMaxPriorityFee(text);
+
+      const newGasParams = {
+        ...selectedGasFee.gasFeeParams,
+        maxPriorityFeePerGas: newMaxPriorityFeePerGas,
+      };
+      updateToCustomGasFee(newGasParams);
+    },
+    [selectedGasFee.gasFeeParams, updateToCustomGasFee]
+  );
+
+  const renderWarning = useCallback(
+    (error, warning) => {
+      if (!selectedOptionIsCustom) return;
+      return (
+        (error && <PanelError>{error}</PanelError>) ||
+        (warning && <PanelWarning>{warning}</PanelWarning>)
+      );
+    },
+    [selectedOptionIsCustom]
+  );
+
+  useEffect(() => {
+    // validate not zero
+    if (!maxBaseFee || maxBaseFee === 0) {
```

Comment:
if using the code changes above (using utilities functions) then these should be converted to using `greaterThan` `multiply` `isZero` etc